src/transaction_manager/transaction_queue.cpp

```diff
@@ -92,43 +92,56 @@ void TransactionQueue::removeTransactionFromBuffer(trx_hash_t const &hash) {
 }
 
 void TransactionQueue::addTransactionToVerifiedQueue(trx_hash_t const &hash, std::list<Transaction>::iterator iter) {
-  uLock lock(shared_mutex_for_verified_qu_);
+  uLock verify_lock(shared_mutex_for_verified_qu_);
+  {
+    // this can happend when tx is verified in backround thread
+    // and in same time we verified block, so it was already removed
```

Comment:
hmm I would not really know what this is about if we did not have personal convo about it...